pyomo/contrib/simplification/build.py

```diff
@@ -0,0 +1,203 @@
+#  ___________________________________________________________________________
+#
+#  Pyomo: Python Optimization Modeling Objects
+#  Copyright (c) 2008-2024
+#  National Technology and Engineering Solutions of Sandia, LLC
+#  Under the terms of Contract DE-NA0003525 with National Technology and
+#  Engineering Solutions of Sandia, LLC, the U.S. Government retains certain
+#  rights in this software.
+#  This software is distributed under the 3-clause BSD License.
+#  ___________________________________________________________________________
+
+import glob
+import logging
+import os
+import shutil
+import sys
+import subprocess
+
+from pyomo.common.download import FileDownloader
+from pyomo.common.envvar import PYOMO_CONFIG_DIR
+from pyomo.common.fileutils import find_library, this_file_dir
+from pyomo.common.tempfiles import TempfileManager
+
+logger = logging.getLogger(__name__ if __name__ != '__main__' else 'pyomo')
+
+
+def build_ginac_library(parallel=None, argv=None, env=None):
+    sys.stdout.write("\n**** Building GiNaC library ****\n")
+
+    configure_cmd = ['configure', '--prefix=' + PYOMO_CONFIG_DIR, '--disable-static']
+    make_cmd = ['make']
+    if parallel:
+        make_cmd.append(f'-j{parallel}')
+    install_cmd = ['make', 'install']
+
+    env = dict(os.environ)
+    pcdir = os.path.join(PYOMO_CONFIG_DIR, 'lib', 'pkgconfig')
+    if 'PKG_CONFIG_PATH' in env:
+        pcdir += os.pathsep + env['PKG_CONFIG_PATH']
+    env['PKG_CONFIG_PATH'] = pcdir
+
+    with TempfileManager.new_context() as tempfile:
+        tmpdir = tempfile.mkdtemp()
+
+        downloader = FileDownloader()
+        if argv:
+            downloader.parse_args(argv)
+
+        url = 'https://www.ginac.de/CLN/cln-1.3.7.tar.bz2'
+        cln_dir = os.path.join(tmpdir, 'cln')
+        downloader.set_destination_filename(cln_dir)
+        logger.info(
+            "Fetching CLN from %s and installing it to %s"
+            % (url, downloader.destination())
+        )
+        downloader.get_tar_archive(url, dirOffset=1)
+        assert subprocess.run(configure_cmd, cwd=cln_dir, env=env).returncode == 0
+        logger.info("\nBuilding CLN\n")
+        assert subprocess.run(make_cmd, cwd=cln_dir, env=env).returncode == 0
+        assert subprocess.run(install_cmd, cwd=cln_dir, env=env).returncode == 0
+
+        url = 'https://www.ginac.de/ginac-1.8.7.tar.bz2'
+        ginac_dir = os.path.join(tmpdir, 'ginac')
+        downloader.set_destination_filename(ginac_dir)
+        logger.info(
+            "Fetching GiNaC from %s and installing it to %s"
+            % (url, downloader.destination())
+        )
+        downloader.get_tar_archive(url, dirOffset=1)
+        assert subprocess.run(configure_cmd, cwd=ginac_dir, env=env).returncode == 0
+        logger.info("\nBuilding GiNaC\n")
+        assert subprocess.run(make_cmd, cwd=ginac_dir, env=env).returncode == 0
+        assert subprocess.run(install_cmd, cwd=ginac_dir, env=env).returncode == 0
```

Comment:
Suggest adding a line at the end of the build:
```
print("Installed GiNaC to %s" % (ginar_dir,))
```

Reply:
Great suggestion. Done.